pkg/crypto/tinkcrypto/crypto.go

```diff
@@ -151,3 +154,35 @@ func (t *Crypto) Verify(sig, msg []byte, kh interface{}) error {
 
 	return err
 }
+
+// ComputeMAC computes message authentication code (MAC) for code data
+// using a matching MAC primitive in kh key handle
+func (t *Crypto) ComputeMAC(data []byte, kh interface{}) ([]byte, error) {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, errBadKeyHandleFormat
+	}
+
+	macPrimitive, err := mac.New(keyHandle)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't know to create the error condition here. I attempted passing an inappropriate key handle (for example, `keyset.NewHandle(aead.AES256GCMKeyTemplate()`), but unfortunately that causes the Tink code to panic instead of returning an err.

The only other thing I can think of is to mac that mac.New function a parameter so that I can pass in a mock for the unit test, but then I'll have to add that to the interface, but I think that would be a bit ugly and perhaps confusing...

Reply:
yes, I noticed that Tink doesn't validate key handles when doing for instance mac.New() or signature.New() and then panics when calling CompuateMac() or Sign() because it tries to convert the keyset to the primitive specific type. I will file an issue in Tink to fix this.

Reply:
Nice, thanks Baha :)